### PR TITLE
add dli queues support

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add dli queues support ; 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X]  Documentation updated.
* [X]  Schema updated.

## Acceptance Steps Performed

make testacc TEST='./flexibleengine' TESTARGS='-run=TestAccDliQueueV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine -v -run=TestAccDliQueueV1_basic -timeout 720m
=== RUN   TestAccDliQueueV1_basic
=== PAUSE TestAccDliQueueV1_basic
=== CONT  TestAccDliQueueV1_basic
--- PASS: TestAccDliQueueV1_basic (20.70s)
PASS
ok      github.com/terraform-providers/terraform-provider-flexibleengine/flexibleengine 20.733s
```

### CODE REVIEW

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+}
+
+func resourceDliQueueV1Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	region := GetRegion(d, config)
+
+	dliClient, err := config.Config.DliV1Client(region)
+
+	if err != nil {
+		return fmt.Errorf("creating sdk client failed, err=%s", err)
```
Comment (chengxiangdong) [inline flexibleengine/resource_flexibleengine_dli_queues_v1.go:187]: which sdk client?  Usually we print `Error creating HuaweiCloud APIG v2 client`

File: flexibleengine/provider.go [hunk 2]
```diff
@@ -344,7 +344,7 @@ func Provider() terraform.ResourceProvider {
 			"flexibleengine_waf_rule_cc_protection":             resourceWafRuleCCAttackProtection(),
 			"flexibleengine_waf_rule_precise_protection":        resourceWafRulePreciseProtection(),
 			"flexibleengine_waf_rule_web_tamper_protection":     resourceWafRuleWebTamperProtection(),
-
+			"flexibleengine_dli_queues":                         ResourceDliQueueV1(),
```
Comment (ShiChangkuo) [inline flexibleengine/provider.go:347]: suggest to rename the resource as `flexibleengine_dli_queue`

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+
+	"github.com/huaweicloud/golangsdk/openstack/common/tags"
+	"github.com/huaweicloud/golangsdk/openstack/dli/v1/queues"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+var regexp4Name = regexp.MustCompile(`^[a-z0-9_]+$`)
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dli_queues_v1.go:16]: suggest to add the max length in regexp

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+					return
+				},
+			},
+
+			"queue_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "sql",
+				ValidateFunc: validation.StringInSlice([]string{"sql", "general"}, false),
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dli_queues_v1.go:43]: according to the API,  the valid option includes sql, general and **all**.

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+
+			"feature": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"basic", "ai"}, false),
+			},
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dli_queues_v1.go:101]: it seems **charging_mode, enterprise_project_id, platform, resource_mode, labels and  feature** are not supported.

File: flexibleengine/resource_flexibleengine_dli_queues_v1_test.go [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+		},
+	})
+}
+
+func testAccDliQueueV1_basic(rName string) string {
+	return fmt.Sprintf(`
+		resource "flexibleengine_dli_queues" "test" {
+ 		 name = "%s"
+		 cu_count              = 16
+		 resource_mode         = 0
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dli_queues_v1_test.go:40]: the foramt is not correct

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go [hunk 7]
```diff
@@ -0,0 +1,261 @@
...
+	}
+
+	//filter by queue_name
+	queueDetail, err := filterByQueueName(queryAllResult.Body, queueName)
+	if err != nil {
+		return err
+	}
+
+	if queueDetail != nil {
+		log.Printf("[debug]The detail of queue from SDK:%+v", queueDetail)
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dli_queues_v1.go:210]: the log level shoud be [DEBUG]

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go [hunk 8]
```diff
@@ -0,0 +1,261 @@
...
+			},
+
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+				ForceNew: true,
+			},
+
+			"cu_count": {
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dli_queues_v1.go:53]: seems this value can be updated by [API](https://docs.prod-cloud-ocb.orange-business.com/api/dli/dli_02_0249.html)

**Review by ShiChangkuo — CHANGES_REQUESTED** [review]

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go (hunk 5 above)
Comment (hmbig2) [reply flexibleengine/resource_flexibleengine_dli_queues_v1.go]: enterprise_project_id,platform,resource_mode are supported, others will been removed

File: flexibleengine/resource_flexibleengine_dli_queues_v1.go (hunk 8 above)
Comment (hmbig2) [reply flexibleengine/resource_flexibleengine_dli_queues_v1.go]: Will be supported in later versions

**Review by ShiChangkuo — APPROVED** [review]
LGTM

**Pull request merged**